### ISSUE TITLE
URLs containing ampersands are broken by excessive escaping

### ISSUE THREAD

**Issue by johnsonm** [elk-zone/elk#1526]
I recently saw a URL to a phpBB forum which uses ampersands in the URLs to delimit arguments. It didn't work, and I initially thought the URL was broken as posted. 

https://main.elk.zone/social.makerforums.info/@chrishuck@fosstodon.org/109774040376522170

![image](https://user-images.githubusercontent.com/79751/215479938-f6c6b3c2-201a-4ee0-88e2-66fee3033048.png)

I finally noticed that the URL had an `&amp;` where a `&` should go, and was able to find the page.

In the default client, the URL is not over-escaped, and the URL works:

https://social.makerforums.info/@chrishuck@fosstodon.org/109774040376522170

![image](https://user-images.githubusercontent.com/79751/215480708-1079b8e4-63f3-4b4d-b1a0-180b241242da.png)

**Comment by rakslice** [comment]
tl;dr for the devs:

1. Put the text `https://foo.bar/baz=1&bat=2` in a toot
2. View the toot in Elk.

Expected: 
Elk's page showing the toot has `<a href="https://foo.bar/baz=1&bat=2" `...

Actual:
Elk's page showing the toot has `<a href="https://foo.bar/baz=1&amp;bat=2" `... , and as a result, following the link takes you to the wrong page
